src/lib/storage/chunk.cpp

```diff
@@ -176,6 +176,25 @@ void Chunk::migrate(boost::container::pmr::memory_resource* memory_source) {
 
 const PolymorphicAllocator<Chunk>& Chunk::get_allocator() const { return _alloc; }
 
+MemoryUsage Chunk::estimate_memory_usage(MemoryUsageEstimationMode estimation_mode) const {
+  auto bytes = size_t{sizeof(*this)};
+
+  for (const auto& column : _columns) {
+    bytes += column->estimate_memory_usage(estimation_mode).bytes;
+  }
+
+  // TODO(anybody) Index memory usage missing
+  // TODO(anybody) AccessCounter memory usage missing
+
+  if (_mvcc_columns) {
+    bytes += _mvcc_columns->tids.size() * sizeof(decltype(_mvcc_columns->tids)::value_type);
```

Comment:
Should this include `sizeof(_mvcc_columns)` and `sizeof(_mvcc_column->tids)` etc.? I know it probably does make a difference, but you also added `sizeof(*this)`.

Reply:
yes